### DISCUSSION TITLE
Redirected to Microsoft Sign in page on Chrome but not Edge

### DISCUSSION THREAD

**Discussion by jy13249** [cypress-io/cypress#24801] [Questions and Help]
So I'm trying to access an internal website to write tests for. Normally when you access the site, you get redirected to login.microsoftonline.com and need to login with your username. Then you will get redirected to the initial page. 

In Cypress using Chrome, I get redirected to the MS page, enter in my username, click Submit, and then get taken to a white page(nothing loads). 

In Cypress in Edge, I don't get redirected to the MS page, and instead just actually go to the internal website. I guess it automatically signed me in(not sure how). Same thing happens with Electron as well.

Has anyone else experienced this before? It seems like ONLY for chrome does it take me to the MS Sign in page, and even when i type in my username and submit, nothing loads. 

I'm not sure if there's some setting I'm missing that I only need to initialize or set for Chrome.

My Cypress.config file looks like this:
`const { defineConfig } = require("cypress");

module.exports = defineConfig({
  e2e: {
    experimentalSessionAndOrigin: true,
    experimentalModifyObstructiveThirdPartyCode: true,
    setupNodeEvents(on, config) {
      // implement node event listeners here
    }
  },
  chromeWebSecurity: false
});`

**Comment by portyu9** [comment]
Hi @jy13249 

I don't think there is a missing Chrome flag here.

The Chrome behavior should be treated as a **cross-origin Microsoft authentication flow**. Current Cypress supports this with `cy.origin()` rather than `chromeWebSecurity: false`.

For Azure AD/Microsoft login, Cypress's current recommended setup is:

```js
module.exports = defineConfig({
  e2e: {
    experimentalModifyObstructiveThirdPartyCode: true,
  },
})
```

Then interact with the Microsoft page inside its origin:

```js
cy.origin('https://login.microsoftonline.com', { args: { username } }, ({ username }) => {
  cy.get('input[type="email"]').type(username)
  cy.get('input[type="submit"]').click()
})
```

Depending on the Microsoft account, the password step can also move to `login.live.com`, which needs its own `cy.origin()` block.

`experimentalSessionAndOrigin` is obsolete; the functionality became stable long ago. `cy.session()` can now be used to cache the authenticated session.

The thread doesn't contain enough information to explain why Edge/Electron went directly into the application — existing SSO/session/environment behavior is possible, but it isn't proven here.

If Chrome still becomes blank after converting the login flow to `cy.origin()`, the next useful evidence is the URL plus browser console/network errors immediately after submitting the Microsoft login.

Please mark this answer as accepted if it helped, thank you.
